cmd/nvidia-dra-controller/imex.go

```diff
@@ -109,7 +116,12 @@ func manageResourceSlices(ctx context.Context, clientset kubernetes.Interface, o
 				delete(newDriverResources.Pools, removedDomain)
 				controller.Update(&newDriverResources)
 				driverResources = newDriverResources
-			case <-ctx.Done():
+			case <-sigs:
+				controller.Stop()
+				err = cleanupImexChannels(ctx, clientset)
+				if err != nil {
+					klog.Errorf("error cleaning up resource slices: %v", err)
+				}
 				return
 			}
```

Comment:
Let's keep the `ctx-Done()` and move the signal handling out to maink with the logic to cleanup the IMEX channels moved into a `Stop()` function on the manager.

That way we can catch the signal, cancel the context, and call stop on the manager all in one place.

Reply:
ok, working on it

Reply:
To do this properly we will also need a waitgroup that blocks the stop from completing until this goroutine completes after the context is canceled.

Reply:
We create the controller in this func `manageResourceSlices` so in order to call `controller.Stop()` we need to be local to this func, or to create the controller and pass it as a value to `manageResourceSlices` (or turn it into a method of the controller)